### ISSUE TITLE
Add version of pyawaitable_await_function that accepts keyword arguments directly.

### ISSUE THREAD

**Issue by AraHaan** [ZeroIntensity/pyawaitable#40]
### Proposal:

```c
pyawaitable_await_function_kwargs(
    PyObject *awaitable,
    PyObject *func,
    PyObject *args,
    const char *fmt,
    awaitcallback cb,
    awaitcallback_err err,
    ...);
```

In this version, the ``vargs`` are used to construct the kwargs that gets passed to ``PyObject_Call``. Alternatively the function could also be used in ``pyawaitable_await_function`` internally with no ``vargs`` passed in to the new function. It can also check and see if the object ``func`` supports the vectorcall protocol and call that using ``PyVectorcall_Call`` instead of ``PyObject_Call`` if it is supported.

The vectorcall protocol can be checked using ``if (PyVectorcall_Function(method) != NULL) {``.

**Comment by AraHaan** [comment]
It seems that ``PyObject_Call`` actually checks if it supports vectorcall internally and then uses it instead if the object does.

**Comment by ZeroIntensity** [comment]
No, I mean that CPython does not have a `PyObject_CallFunction` variation that accepts kwargs. I don't think it fits in PyAwaitable if it doesn't exist upstream.
